logic/scanning_logic.py

```diff
@@ -0,0 +1,629 @@
+                    self._current_dummy_data = self._generate_2d_dummy_data(scan_axes)
+                else:
+                    self._current_dummy_data = self.generate_1d_dummy_data(scan_axes[0])
+                self.__scan_line_count = 0
+                self.__scan_start_time = time.time()
+                self._scan_data[self.__running_scan] = ScanData(
+                    scan_axes=self.__running_scan,
+                    channel_config=self.scanner_constraints['data_channels'],
+                    scanner_settings=self.scanner_settings)
+                self._scan_data[self.__running_scan].new_data()
+                num_x_vals = self.scanner_settings['scan_resolution'][scan_axes[0]]
+                if len(scan_axes) > 1:
+                    self.__scan_line_interval = num_x_vals / self.scanner_settings[
+                        'pixel_clock_frequency']
+                    self.__scan_line_positions = {ax: np.full(num_x_vals, self.scanner_target[ax])
+                                                  for
+                                                  ax in self._constraints['axes']}
+                    min_val, max_val = self.scanner_settings['scan_range'][self.__running_scan[0]]
+                    self.__scan_line_positions[self.__running_scan[0]] = np.linspace(min_val,
+                                                                                     max_val,
+                                                                                     num_x_vals)
+                else:
+                    self.__scan_line_interval = 10 / self.scanner_settings['pixel_clock_frequency']
+                    self.__scan_line_positions = self.scanner_target.copy()
+
+                self.__scan_stop_requested = False
+                self.__sigNextLine.emit()
+            else:
+                self.__scan_stop_requested = True
+        return
+
+    @QtCore.Slot()
+    def _scan_loop(self):
+        if self.module_state() != 'locked':
+            return
+
+        with self.threadlock:
+            if len(self.__running_scan) > 1:
+                max_number_of_lines = self.scanner_settings['scan_resolution'][self.__running_scan[1]]
+            else:
+                max_number_of_lines = self.scanner_settings['scan_resolution'][self.__running_scan[0]]
+            if self.__scan_line_count >= max_number_of_lines or self.__scan_stop_requested:
+                while len(self._history) >= self.max_history_length:
+                    self._history.pop(0)
+                self._history.append(copy.deepcopy(self.scan_data[self.__running_scan]))
+                self._history_index = len(self._history) - 1
+                self.module_state.unlock()
+                self.sigScanStateChanged.emit(False, self.__running_scan)
+                self.__timer.start()
+                return
+
+            if len(self.__running_scan) > 1:
+                y_min, y_max = self.scanner_settings['scan_range'][self.__running_scan[1]]
```

Comment:
by working with self.__running_scan[-1] you can get around the whole business of distinguishing
`if len(self.__running_scan) > 1:`

Reply:
Ok, how?
I see how 
```
x_min, x_max = self.scanner_settings['scan_range'][self.__running_scan[-1]]
```
will work but the assignment of
```
self.__scan_line_positions[self.__running_scan[1]] = ...
```
is still different.

Reply:
Ah damn, yes I did not pay attention to the fact that you are using `scan_resolution` in one case and `__scan_line_positions` in the other.